sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -1,43 +1,457 @@
+            selector: 'slate-tasks-teacher-taskassigner',
+            autoCreate: true,
+
+            xtype: 'slate-tasks-teacher-taskassigner'
+        },
+        taskRater: {
+            selector: 'slate-tasks-teacher-taskrater',
+            autoCreate: true,
+
+            xtype: 'slate-tasks-teacher-taskrater'
+        },
+        taskEditor: {
+            selector: 'slate-tasks-teacher-taskeditor',
+            autoCreate: true,
+
+            xtype: 'slate-tasks-teacher-taskeditor'
+        },
+        taskEditorForm: 'slate-tasks-teacher-taskeditor slate-modalform',
+        skillsField: 'slate-tasks-teacher-taskeditor slate-skillsfield',
+        attachmentsField: 'slate-tasks-teacher-taskeditor slate-tasks-attachmentsfield',
+        assignmentsField: 'slate-tasks-teacher-taskeditor slate-tasks-assignmentsfield',
+        assignmentsComboField: 'slate-tasks-teacher-taskeditor slate-tasks-assignmentsfield combo',
+
+        courseSelector: 'slate-tasks-teacher-appheader combo'
+    },
 
     // controller templates method overrides
     onLaunch: function () {
         this.getDashboardCt().render('slateapp-viewport');
     },
 
-
     // event handlers
-    onCompetencyRowClick: function(me, competency, ev, targetEl) {
-        me.toggleCompetency(competency);
+    onCourseSectionSelect: function(combo, record) {
+        var me = this;
+
+        me.getDashboardCt().setCourseSection(record);
+    },
+
+    onDashboardSectionChange: function(record) {
+        this.redirectTo('section/'+record.get('Code'));
+    },
+
+    onAcceptTaskClick: function() {
+        var me = this,
+            taskRater = me.getTaskRater(),
+            studentTask = taskRater.getStudentTask();
+
+        studentTask.set('TaskStatus', 'completed');
+
+        me.saveStudentTask(studentTask, function() {
+            taskRater.close();
+        });
+    },
+
+    onAssignRevisionClick: function(btn) {
+        var me = this,
+            coords = btn.getXY(),
+            datepicker;
+
+        if (btn.dateSelected) {
+            me.assignStudentTaskRevision(btn.dateSelected);
+        } else {
+            datepicker = Ext.widget({
+                xtype: 'datepicker',
+                floating: true,
+                handler: function(picker, date) {
+                    picker.destroy();
+                    btn.dateSelected = date;
+                    return btn.setText('Re-Assign revision due on '+Ext.Date.format(date, 'm/d/y'));
+                }
+            });
+            datepicker.show().showAt(coords[0], coords[1] - datepicker.getHeight()); // subtract datepicker height from y coord to show above button.
+        }
+    },
+
+    onUnassignStudentTaskClick: function() {
+        var me = this,
+            taskRater = me.getTaskRater(),
+            studentTask = taskRater.getStudentTask();
+
+        Ext.Msg.confirm('Are you sure?', 'Do you want to unassign this student task? This can not be undone.', function(ans) {
```

Comment:
@nbey `Ext.MessageBox` needs to be explicitly listed in this class's `requires` if you're going to use the `Ext.Msg` singleton it provides